### PR TITLE
Feat/edit diary entry in details

### PR DESCRIPTION
**Pull request by AnsMann**

### CODE REVIEW

File: src/app/DiaryEntryDetails.js [hunk 1]
```diff
@@ -68,42 +77,85 @@ export function DiaryEntryDetails({
...
       content: coachFeedback,
+      type: 'coachFeedback',
     },
     {
       headline: 'Außerdem war mir heute noch wichtig',
       content: additional,
+      type: 'additional',
     },
   ]
+  const [editRating, setEditrating] = useState(false)
```
Comment (melmare) [inline src/app/DiaryEntryDetails.js]: Here is the question similar to my last code review for the "showDetail"-State if you remember. You use a Boolean, so maybe a State-Naming which indicates that like "isRatingEditable" or even just "isEditable" could be an idea.

File: src/app/DiaryEntryDetails.js [hunk 2]
```diff
@@ -68,42 +77,85 @@ export function DiaryEntryDetails({
...
+  function handleDayRatingClick() {
+    setEditrating(true)
+  }
+
+  function handleEditDetails(detailType, input) {
+    onEditDetails(id, detailType, input)
+  }
+
+  function handleEditRating(event) {
+    const { target } = event
```
Comment (melmare) [inline src/app/DiaryEntryDetails.js:119]: I it really necessary to destructure the target here? You are just saving one word by write a complete line. You could consider `const form = event.target` or something like that.

File: src/app/DiaryEntryDetails.js [hunk 3]
```diff
@@ -68,42 +77,85 @@ export function DiaryEntryDetails({
...
+            key={detailObject.headline}
+            title={detailObject.headline}
+            content={detailObject.content}
+            onEditDetail={handleEditDetails}
+            detailType={detailObject.type}
+          />
         ))}
-        <ShowDayRating entryRating={rating} />
+        {editRating ? (
+          <OutsideClickHandler onOutsideClick={() => setEditrating(false)}>
```
Comment (melmare) [inline src/app/DiaryEntryDetails.js]: Great idea to use that OutsideClickHandler-Component!

File: src/app/ShowSingleDetail.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        value={content}
+        name={detailType}
+        onEditText={onEditDetail}
+      />
+    </OutsideClickHandler>
+  ) : (
+    <SingleDetail>
+      <h3>{title}</h3>
+      <p onClick={() => setIsEditable(true)}>
+        {content ? content : 'Kein Eintrag'}
```
Comment (melmare) [inline src/app/ShowSingleDetail.js]: You could use `{content || 'Kein Eintrag'}` here.

File: src/app/DiaryEntryDetails.js (hunk 1 above)
Comment (AnsMann) [reply src/app/DiaryEntryDetails.js]: changed

File: src/app/ShowSingleDetail.js (hunk 4 above)
Comment (AnsMann) [reply src/app/ShowSingleDetail.js]: good points thanks :)

File: src/app/DiaryEntryDetails.js (hunk 2 above)
Comment (AnsMann) [reply src/app/DiaryEntryDetails.js:119]: i think i keep it here :)

**Pull request merged**